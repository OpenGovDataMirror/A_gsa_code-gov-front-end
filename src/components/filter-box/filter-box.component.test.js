import React from 'react'
import { shallow } from 'enzyme'

import FilterBox from 'components/filter-box/filter-box.component'

const props = {
  title: 'test-title',
  options: JSON.stringify([
    { id: 1, value: 'opt-1' },
    { id: 2, value: 'opt-2' },
    { id: 3, value: 'opt-3' },
    { id: 4, value: 'opt-4' },
    { id: 5, value: 'opt-5' }
  ]),
  onChange: jest.fn()
}

let wrapper
let instance
describe('components - FilterBox', () => {
  beforeEach(() => {
    wrapper = shallow(<FilterBox {...props} />)
    instance = wrapper.instance()
  })

  describe('onChange', () => {
    it('should be called for checkbox input onChange', () => {
      const event = {
        target: { checked: false, value: 'test-value', tagName: 'INPUT' }
      }
      const input = wrapper.find('input[value="opt-1"]')
      input.simulate('change', event)
      expect(props.onChange).toBeCalledWith({ type: 'unchecked', value: 'test-value' })
    })
  })

  describe('handleKeyDown', () => {
    it('should inverse the value of showAll state if enter key is pressed', () => {
      const event = {
        keyCode: 13
      }
      wrapper.setState({ showAll: false })

      // false => true
      instance.handleKeyDown(event)
      expect(wrapper.state('showAll')).toBeTruthy()

      // true => false
      instance.handleKeyDown(event)
      expect(wrapper.state('showAll')).toBeFalsy()
    })
    it('should call preventDefault if space bar is pressed', () => {
      const event = {
        keyCode: 32,
        preventDefault: jest.fn()
      }
      instance.handleKeyDown(event)
      expect(event.preventDefault).toBeCalled()
    })
  })

  describe('handleKeyUp', () => {
    it('should inverse the value of showAll state if space bar is pressed', () => {
      const event = {
        keyCode: 32
      }
      wrapper.setState({ showAll: false })

      // false => true
      instance.handleKeyUp(event)
      expect(wrapper.state('showAll')).toBeTruthy()

      // true => false
      instance.handleKeyUp(event)
      expect(wrapper.state('showAll')).toBeFalsy()
    })
  })

  describe('handleClickToggle', () => {
    it('should inverse the value of showAll state if moreLessToggle is clicked', () => {
      const event = {
        preventDefault: jest.fn()
      }
      wrapper.setState({ showAll: false })

      // false => true
      instance.handleClickToggle(event)
      expect(wrapper.state('showAll')).toBeTruthy()

      // true => false
      instance.handleClickToggle(event)
      expect(wrapper.state('showAll')).toBeFalsy()
    })
  })

  describe('handleExpand', () => {
    it('should inverse the value of accordionExpand state if the accordion title button is clicked', () => {
      wrapper.setState({ accordionExpand: true })
      // true => false
      instance.handleExpand()
      expect(wrapper.state('accordionExpand')).toBeFalsy()

      // false => true
      instance.handleExpand()
      expect(wrapper.state('accordionExpand')).toBeTruthy()
    })
  })
  describe('render', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
