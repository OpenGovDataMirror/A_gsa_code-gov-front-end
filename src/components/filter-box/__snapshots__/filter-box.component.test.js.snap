// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - FilterBox render should render correctly 1`] = `
<div
  className="usa-accordion border-base-light border-1px tablet-lg:margin-bottom-2 margin-bottom-0"
>
  <div
    className="usa-accordion__heading"
  >
    <button
      aria-controls="test-title"
      aria-expanded={true}
      className="usa-accordion__button font-heading-md text-base-darker"
      id="test-title-button"
      onClick={[Function]}
    >
      test-title
    </button>
  </div>
  <div
    className="usa-accordion__content usa-prose border-top-1px border-base-light padding-bottom-105"
    hidden={false}
    id="test-title"
  >
    <form
      className="usa-form maxw-none"
    >
      <fieldset
        aria-labelledby="test-title-button"
        className="usa-fieldset"
      >
        <legend
          aria-labelledby="test-title-button"
          className="usa-sr-only"
        />
        <div
          className="usa-checkbox"
        >
          <input
            className="usa-checkbox__input"
            id="opt-1"
            onChange={[Function]}
            type="checkbox"
            value="opt-1"
          />
          <label
            className="usa-checkbox__label font-body-2xs text-base-dark"
            htmlFor="opt-1"
          >
            <span />
          </label>
        </div>
        <div
          className="usa-checkbox"
        >
          <input
            className="usa-checkbox__input"
            id="opt-2"
            onChange={[Function]}
            type="checkbox"
            value="opt-2"
          />
          <label
            className="usa-checkbox__label font-body-2xs text-base-dark"
            htmlFor="opt-2"
          >
            <span />
          </label>
        </div>
        <div
          className="usa-checkbox"
        >
          <input
            className="usa-checkbox__input"
            id="opt-3"
            onChange={[Function]}
            type="checkbox"
            value="opt-3"
          />
          <label
            className="usa-checkbox__label font-body-2xs text-base-dark"
            htmlFor="opt-3"
          >
            <span />
          </label>
        </div>
        <div
          className="usa-checkbox"
        >
          <input
            className="usa-checkbox__input"
            id="opt-4"
            onChange={[Function]}
            type="checkbox"
            value="opt-4"
          />
          <label
            className="usa-checkbox__label font-body-2xs text-base-dark"
            htmlFor="opt-4"
          >
            <span />
          </label>
        </div>
        <div
          className="usa-checkbox"
          hidden={false}
        >
          <input
            className="usa-checkbox__input"
            id="opt-5"
            onChange={[Function]}
            type="checkbox"
            value="opt-5"
          />
          <label
            className="usa-checkbox__label font-body-2xs text-base-dark"
            htmlFor="opt-5"
          >
            <span />
          </label>
        </div>
      </fieldset>
    </form>
    <div>
      <span
        className="moreLessToggle text-underline text-primary font-body-2xs"
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        tabIndex="0"
      >
        Show less
      </span>
    </div>
  </div>
</div>
`;
